File: java/src/main/java/GearBox.java


public class GearBox {

    private int gear = 0;

    public void doit(int i) {
        if (gear > 0) {
            if (i > 2000)
                gear++;
            else if (i < 500) {
                gear--;
            }
        }
        if (gear > 6) {
            gear--;
        } else if (gear < 1) {
            gear++;
        }
    }

    public int getCurrentGear() {
        return gear;
    }

    public void setGearRange(int gear, int shiftDown, int shiftUp) {
        // TODO: Implement Me!
    }
}

File: c++/GearBox.cpp
#include "GearBox.h"

GearBox::GearBox() : s(0), e(0) {

}

void GearBox::doit(int i) {
    if (s < 0) {
        // do nothing
        e = i;
    } else {
        if (s > 0) {
            if (i > 2000)
                s++;
        } else if (i < 500) {
            s--;
        }
    }
    if (s > 6) {
        s--;
    } else if (s < 1) {
        s++;
    }
    e = 1;
}

int GearBox::currentGear(void) {
    return s;
}


